pkg/share/manager/jsoncs3/jsoncs3.go

```diff
@@ -498,6 +496,10 @@ func (m *Manager) UpdateShare(ctx context.Context, ref *collaboration.ShareRefer
 	m.Lock()
 	defer m.Unlock()
 
+	user := ctxpkg.ContextMustGetUser(ctx)
+
+	m.invalidateCache(ctx, user)
```

Comment:
Question: Do we only use `invalidateCache` when updating a share? This could be a problem because as far as I see this is only place where expired shares are removed.

Reply:
Moved it to every writing operation.